src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -452,28 +458,37 @@ def _make_time_value(value: XMLValue) -> dict[str, Any]:
     }
 
 
-def _serialise_uri_prop(prop: XMLProperty, permissions_lookup: dict[str, Permissions]) -> dict[str, Any]:
-    uri_values: list[SerialiseValue] = []
-    for v in prop.values:
-        if v.permissions:
-            if not (per := permissions_lookup.get(v.permissions)):
-                raise PermissionNotExistsError(f"Could not find permissions for value: {v.permissions}")
-            permission = str(per)
-        else:
-            permission = None
-        uri = cast(str, v.value)
-        uri_values.append(
-            SerialiseURI(
-                value=uri,
-                permissions=permission,
-                comment=v.comment,
-            )
-        )
+def _transform_into_serialise_property_with_string_value(
+    prop: XMLProperty,
+    permissions_lookup: dict[str, Permissions],
+    func: Callable[[str, str | None, str | None], SerialiseValue],
+) -> SerialiseProperty:
+    serialised_values: list[SerialiseValue] = [
+        _transform_into_serialise_value_with_string_value(v, permissions_lookup, func) for v in prop.values
+    ]
```

Comment:
1. The name is a bit complicated. What about `_serialise_val()`?
2. The type hint is obsolete

```suggestion
    serialised_values = [_serialise_val(v, permissions_lookup, serialiser) for v in prop.values]
```

Reply:
Regarding the name same as above.